reconcile/external_resources/secrets_sync.py

```diff
@@ -217,13 +244,41 @@ def build_incluster_secrets_reconciler(
         integration=QONTRACT_INTEGRATION,
     )
     oc = ocmap.get_cluster(cluster)
-    vault_client = VaultClient()
     return InClusterSecretsReconciler(
         cluster=cluster,
         namespace=namespace,
         ri=ri,
         oc=oc,
         vault_path=vault_path,
-        vault_client=vault_client,
+        vault_client=VaultClient(),
         secrets_reader=secrets_reader,
+        thread_pool_size=thread_pool_size,
+        dry_run=dry_run,
     )
+
+
+class VaultSecretsReconciler(SecretsReconciler):
+    def __init__(
+        self,
+        ri: ResourceInventory,
+        secrets_reader: SecretReaderBase,
+        vault_path: str,
+        thread_pool_size: int,
+        dry_run: bool,
+    ):
+        super().__init__(ri, secrets_reader, thread_pool_size, dry_run)
+        self.secrets_reader = secrets_reader
+        self.vault_path = vault_path
+
+    def _populate_secret_data(self, specs: Iterable[ExternalResourceSpec]) -> None:
+        threaded.run(self._read_secret, specs, self.thread_pool_size)
+
+    def _read_secret(self, spec: ExternalResourceSpec) -> None:
+        secret_path = f"{self.vault_path}/{spec.cluster_name}/{spec.namespace_name}/{spec.identifier}"
+        try:
+            logging.debug("Reading Secret %s", secret_path)
+            data = self.secrets_reader.read_all({"path": secret_path})
+            spec.secret = data
+        except Exception:
+            msg = f"Error getting secret from vault, skipping. [{secret_path}]"
+            logging.debug(msg)
```

Comment:
This exception only log debug, how do we get notified if some secret is always missing in vault?

Reply:
Some modules might not have outputs, hence no secrets in Vault. New specs might have missing Secrets until they are reconciled for the first time. 
I don't see any reason why a Secret can be always missing but you might be right. Would you prefer this as info level? 

* I'm thinking of adding a module configuration attribute to indicate if a module needs to reconcile secrets or not and avoid those in the loop (that solves a part of the problem).

Reply:
+1 to have config so we can tell if a spec has output/secret, until then we can explicit check secret not exist exception instead of all `Exception`, do info log so we can see if it's always missing.

Reply:
Added the configuration + changed the exception. PTAL.